src/backend/commands/tablecmds.c

```diff
@@ -14825,23 +14825,12 @@ prebuild_temp_table(Relation rel, RangeVar *tmpname, DistributedBy *distro, List
 		{
 			if (useExistingColumnAttributes)
 			{
-				/*
-				 * Need to remove table level compression settings for the
-				 * AOCO case since they're set at the column level.
-				 */
 				ListCell *lc;
 
 				foreach(lc, opts)
```

Comment:
Good job overall.
It seems we can optimize this part of code a bit. We could collapse two `foreach` loops into one keeping `if` condition only for `RelationGetUntransformedAttributeOptions()` call.

Reply:
yes, but it produce more change to original code

Reply:
True, but I don't think that producing copypaste in such case is better.